src/uproot/behaviors/TBranch.py

```diff
@@ -689,6 +689,229 @@ def foreach(start):
+    if "num_fallback_workers" not in real_options:
+        real_options["num_fallback_workers"] = 1
+
+    filter_branch = uproot._util.regularize_filter(filter_branch)
+
+    hasbranches = []
+    common_keys = None
+    is_self = []
+
+    count = 0
+    for file_path, object_path in files:
+        obj = _regularize_object_path(
+            file_path, object_path, custom_classes, allow_missing, real_options
+        )
+
+        if obj is not None:
+            count += 1
+
+            if isinstance(obj, TBranch) and len(obj.keys(recursive=True)) == 0:
+                original = obj
+                obj = obj.parent
+                is_self.append(True)
+
+                def real_filter_branch(branch):
+                    return branch is original and filter_branch(branch)
+
+            else:
+                is_self.append(False)
+                real_filter_branch = filter_branch
+
+            hasbranches.append(obj)
+
+            new_keys = obj.keys(
+                recursive=recursive,
+                filter_name=filter_name,
+                filter_typename=filter_typename,
+                filter_branch=real_filter_branch,
+                full_paths=full_paths,
+            )
+
+            if common_keys is None:
+                common_keys = new_keys
+            else:
+                new_keys = set(new_keys)
+                common_keys = [key for key in common_keys if key in new_keys]
+
+    if count == 0:
+        raise ValueError(
+            "allow_missing=True and no TTrees found in\n\n    {}".format(
+                "\n    ".join(
+                    "{"
+                    + "{}: {}".format(
+                        repr(f.file_path if isinstance(f, HasBranches) else f),
+                        repr(f.object_path if isinstance(f, HasBranches) else o),
+                    )
+                    + "}"
+                    for f, o in files
+                )
+            )
+        )
+
+    if len(common_keys) == 0 or not (all(is_self) or not any(is_self)):
+        raise ValueError(
+            "TTrees in\n\n    {}\n\nhave no TBranches in common".format(
+                "\n    ".join(
+                    "{"
+                    + "{}: {}".format(
+                        repr(f.file_path if isinstance(f, HasBranches) else f),
+                        repr(f.object_path if isinstance(f, HasBranches) else o),
+                    )
+                    + "}"
+                    for f, o in files
+                )
+            )
+        )
+
+    dask_dict = {}
+    for key in common_keys:
+        dask_arrays = []
+        for ttree in hasbranches:
+            entry_start, entry_stop = _regularize_entries_start_stop(
+                ttree.tree.num_entries, None, None
+            )
+            entry_step = 0
+            if uproot._util.isint(step_size):
+                entry_step = step_size
+            else:
+                entry_step = ttree[key].num_entries_for(step_size)
```

Comment:
This is how I have used `step_size` and `TBranch.num_entries_for`

Reply:
On further research, `TBranch.num_enteries_for` always gives 1 as the answer, but `TTree.num_enteries_for` seems to give the correct answer. However, I believe `TTree.num_enteries_for` wouldn't be ideal because if the user has input a memory size, they would probably expect each chunk to be of that memory size.